project/MimaExcludes.scala

```diff
@@ -118,79 +96,23 @@ object MimaExcludes {
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.execution.*"),
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.internal.*"),
     ProblemFilters.exclude[Problem]("org.apache.spark.sql.errors.*"),
+    // SPARK-40283: add jdbc.JdbcDialect as default excludes
+    ProblemFilters.exclude[Problem]("org.apache.spark.sql.jdbc.JdbcDialect.*"),
```

Comment:
`JdbcDialect` is a developer API and should be tracked. Can we track it explicitly instead of skipping the entire package?

Reply:
[34fb8c4](https://github.com/apache/spark/pull/37741/commits/34fb8c4d2e56cb3244ce8c1ffa7cb927fbe4c204) fix this